File: bro-agent-api/src/main/python/bro_agent/agent/BackupExecutionActions.py
import bro_agent.generated.Action_pb2 as Action
from bro_agent.service.BackupService import BackupService


class BackupExecutionActions:
    """ BackupExecutionActions """
    def __init__(self, agent, action_information):
        self.agent = agent
        self.action_information = action_information

    def send_backup(self, fragment_information):
        """ Send Backup """
        backup_service = BackupService(self.agent.channel)
        backup_service.backup(fragment_information,
                              self.agent.get_agent_id(),
                              self.action_information.backup_name)

    def backup_complete(self, success, message):
        self.agent.send_stage_complete_message(success, message, Action.BACKUP)
        self.agent.backup_name = None
        self.agent.backup_type = None

    def get_backup_name(self):
        return self.action_information.backup_name

    def get_backup_type(self):
        return self.action_information.backup_type

File: bro-agent-api/src/main/cpp/agent/BackupExecutionActions.cc
#include "service/BackupService.hh"

#include "Agent.hh"
#include "BackupExecutionActions.hh"

namespace BackupRestoreAgent
{

BackupExecutionActions::BackupExecutionActions(Agent* agent)
    : agent(agent)
{
}

void BackupExecutionActions::sendBackup(BackupFragmentInformation fragmentInformation)
{
    std::cerr << "BackupExecutionActions::sendBackup" << std::endl;
    grpc::ClientContext context;
    BackupService backupService(std::move(agent->getBackupStream(&context)), new FileChunkServiceUtil(fragmentInformation.getBackupFilePath()));
    backupService.backup(fragmentInformation, agent->getAgentId(), agent->getBackupName());
}

void BackupExecutionActions::backupComplete(bool success, const std::string& message)
{
    std::cerr << "BackupExecutionActions::backupComplete" << std::endl;
    agent->sendStageCompleteMessage(success, message, Action::BACKUP);
    agent->clearBackupName();
}

std::string BackupExecutionActions::getBackupName()
{
    return agent->getBackupName();
}
}
